homeassistant/components/sector_alarm/sensor.py

```diff
@@ -0,0 +1,62 @@
+"""The Sector Alarm Integration."""
+from datetime import timedelta
+import logging
+
+from homeassistant.components.sector_alarm import DOMAIN as SECTOR_DOMAIN
+from homeassistant.const import TEMP_CELSIUS
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+UPDATE_INTERVAL = timedelta(minutes=5)
+
+DEPENDENCIES = ["sector_alarm"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Initialition of the platform."""
+    sector_connection = hass.data.get(SECTOR_DOMAIN)
+
+    temps = sector_connection.GetTemps()
+
+    dev = []
+
+    for temp in temps:
+        dev.append(SectorTempSensor(temp, sector_connection))
+
+    async_add_entities(dev)
+
+
+class SectorTempSensor(Entity):
+    """Secotor Alarm temperature class."""
+
+    def __init__(self, temp, sectorconnection):
+        """Initialize the sensor."""
+        self._sector = sectorconnection
+        self._state = None
+        self._name = temp[0]
+        self._state = temp[1]
```

Comment:
Extract these values and then pass them as individual parameters to the constructor to make it more readable.